coalib/settings/Setting.py

```diff
@@ -296,3 +297,26 @@ def value(self):
                              'incomplete. Please access the value of the '
                              'setting in a section to get the complete value.')
         return self._value
+
+    @property
+    def origin(self):
+        """
+        Ensures that Setting.origin always returns a string,
+        i.e. the filename.
+        """
+        if isinstance(self._origin, SourcePosition):
+            return self._origin.filename
+        else:
+            return self._origin
+
+    @origin.setter
+    def origin(self, origin):
+        self._origin = origin
+
+    @property
+    def line_number(self):
+        if isinstance(self._origin, SourcePosition):
```

Comment:
Can be just me, but I personally think `self.origin` is better here. Assume you add some validation in `origin.setter`, but if some bad guy sets `self._origin` explicitly, they can bypass the validation because you are using `self._origin` here.

Reply:
@li-boxuan  Even i think it would be better to do it your way. @jayvdb  What do you think?

Reply:
ya, agreed,

```py
source_position = self.origin
if isinstance(...)
    ...
```

Reply:
Can't use `self.origin` as getter is set to return `str`

Reply:
Oh yep, my fault. Nvm then.